src/bin/nydus-image/inspect.rs

```diff
@@ -361,32 +442,132 @@ Blob ID: {}
-        self.rafs_meta.walk_directory::<PathBuf>(
-            self.rafs_meta.superblock.root_ino(),
-            None,
-            &mut |inode, path| {
-                if inode.ino() == ino {
-                    println!(r#"{}"#, path.to_string_lossy(),);
-                    self.stat_single_file(ino)?;
+    fn cmd_check_inode(&mut self, ino: u64) -> Result<Option<Value>, anyhow::Error> {
+        let current_inode = self.rafs_meta.superblock.get_inode(ino, false)?;
+        if self.rafs_meta.meta.is_v6() && !current_inode.is_dir() {
+            if self.files_parent.is_empty() {
+                self.generate_files_parent()?;
+            }
+
+            if let Some(parents) = self.files_parent.get(&ino) {
+                for parent in parents {
+                    let parent_inode = self
+                        .rafs_meta
+                        .superblock
+                        .get_extended_inode(*parent, false)?;
+                    let parent_path = self.rafs_meta.path_from_ino(*parent)?;
+                    let child_count = parent_inode.get_child_count();
+                    for idx in 0..child_count {
+                        let child = parent_inode.get_child_by_index(idx)?;
+                        if child.ino() == ino {
+                            let path = parent_path.join(child.name());
+                            println!(r#"{}"#, path.to_string_lossy(),);
+                            self.stat_single_file(
+                                Some(parent_inode.as_ref()),
+                                current_inode.as_ref(),
+                            )?;
+                            break;
+                        }
+                    }
                 }
-                Ok(())
-            },
-        )?;
+            }
+        } else {
+            self.walk_dir(
+                self.rafs_meta.superblock.root_ino(),
+                None,
+                None,
+                &mut |parent, inode, path| {
+                    if inode.ino() == ino {
+                        println!(r#"{}"#, path.to_string_lossy(),);
+                        self.stat_single_file(parent, inode)?;
+                    }
+                    Ok(())
+                },
+            )?;
+        }
+
         Ok(None)
     }
 }
 
 impl RafsInspector {
+    /// Get file name of the inode, the rafs v6 file is handled separately.
+    fn get_file_name(&self, parent_inode: &dyn RafsInodeExt, inode: &dyn RafsInode) -> OsString {
+        let mut filename = OsString::from("");
+        if self.rafs_meta.meta.is_v6() && !inode.is_dir() {
+            parent_inode
+                .walk_children_inodes(
+                    0,
+                    &mut |_inode: Option<Arc<dyn RafsInode>>, name: OsString, cur_ino, _offset| {
+                        if cur_ino == inode.ino() {
+                            filename = name;
+                        }
+                        Ok(RafsInodeWalkAction::Continue)
```

Comment:
should return break action thus to exit from the loop ?

Reply:
Yes, fixed it.